qcengine/programs/qchem.py

```diff
@@ -249,13 +255,390 @@ def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") ->
+        """
+
+        NUMBER = r"(?x:" + regex.NUMBER + ")"
+
+        properties = {}
+        provenance = provenance_stamp(__name__)
+        mobj = re.search(r"This is a multi-thread run using ([0-9]+) threads", outtext)
+        if mobj:
+            provenance["nthreads"] = int(mobj.group(1))
+
+        mobj = re.search(r"Total job time:\s*" + NUMBER + r"s\(wall\)", outtext)
+        if mobj:
+            provenance["wall_time"] = float(mobj.group(1))
+
+        mobj = re.search(r"Archival summary:\s*\n[0-9]+\\[0-9+]\\([\w\.\-]+)\\", outtext)
+        if mobj:
+            provenance["hostname"] = mobj.group(1)
+
+        mobj = re.search(r"\n\s*There are\s+(\d+) alpha and\s+(\d+) beta electrons\s*\n", outtext)
+        if mobj:
+            properties["calcinfo_nalpha"] = int(mobj.group(1))
+            properties["calcinfo_nbeta"] = int(mobj.group(2))
+
+        mobj = re.search(r"\n\s*There are\s+\d+ shells and\s+(\d+) basis functions\s*\n", outtext)
+        if mobj:
+            properties["calcinfo_nbasis"] = int(mobj.group(1))
+
+        mobj = re.search(r"\n\s*RI-MP2 CORRELATION ENERGY\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        if mobj:
+            properties["mp2_correlation_energy"] = float(mobj.group(1))
+
+        mobj = re.search(r"\n\s*RI-MP2 SINGLES ENERGY\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        if mobj:
+            properties["mp2_singles_energy"] = float(mobj.group(1))
+
+        mobj_aaaa = re.search(r"\n\s*RI-MP2 ENERGY \(aa\|aa\)\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        mobj_bbbb = re.search(r"\n\s*RI-MP2 ENERGY \(bb\|bb\)\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        if mobj_aaaa and mobj_bbbb:
+            properties["mp2_same_spin_correlation_energy"] = float(mobj_aaaa.group(1)) + float(mobj_bbbb.group(1))
+
+        mobj_aabb = re.search(r"\n\s*RI-MP2 ENERGY \(aa\|bb\)\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        mobj_bbaa = re.search(r"\n\s*RI-MP2 ENERGY \(bb\|aa\)\s+=\s+" + NUMBER + r"\s+au\s*\n", outtext)
+        if mobj_aaaa and mobj_bbbb:
+            properties["mp2_opposite_spin_correlation_energy"] = float(mobj_aabb.group(1)) + float(mobj_bbaa.group(1))
+
+        properties["calcinfo_natom"] = len(input_dict["molecule"]["symbols"])
+
+        mobj = re.search(r"\n\s*(\d+)\s+" + NUMBER + "\s+" + NUMBER + r"\s+Convergence criterion met\s*\n", outtext)
+        if mobj:
+            properties["scf_iterations"] = int(mobj.group(1))
+
+        mobj = re.search(
+            r"\n\s+Dipole Moment \(Debye\)\s*\n\s+X\s+" + NUMBER + r"\s+Y\s+" + NUMBER + r"\s+Z\s+" + NUMBER + r"\s*\n",
+            outtext,
+        )
+        if mobj:
+            cf = constants.conversion_factor("debye", "e * bohr")
+            properties["scf_dipole_moment"] = [float(mobj.group(i)) * cf for i in range(1, 4)]
+
+        return properties, provenance
+
+    def parse_logfile(self, outfiles: Dict[str, str]) -> AtomicResult:
```

Comment:
might be good to throw if qchem link0 encountered (or whatever lets them run multiple jobs per input if link0 is gaussian syntax)